### PR TITLE
Add filter logic to export_records

### PR DESCRIPTION
**Pull request by erikh360**
This adds a `filter_logic` argument to the `export_records` function on the `Project` object.

Unfortunately I don't have access to a test project/token to update the tests.

### CODE REVIEW

**Comment by sburns** [comment]
Thanks for your PR! I think this is your first on PyCap?

> Unfortunately I don't have access to a test project/token to update the tests.

me neither 😢 . I could see improving our tests to do one of two things:

1. Find another institution where we can host an example project. Still brittle because it's going to be owned by someone, they'll eventually leave, etc.
2. Rewrite our tests to not need the internet.

Would you like to do either one?

**Comment by erikh360** [comment]
Thanks for your reply @sburns. Yes my first one!

Unfortunately, I don't have direct access to a instance that can host a example project. How much work would it be to set one up?

Time is also not really on my side to rewrite all the tests. I also haven't checked how much work this would be.

File: test/test_project.py [hunk 1]
```diff
@@ -132,23 +396,25 @@ def test_file_export(self):
         with self.assertRaises(ValueError):
             self.reg_proj.export_file(record=record, field='dob')
         # Delete and make sure we get an RedcapError with next export
-        self.reg_proj.delete_file(record, field)
-        with self.assertRaises(RedcapError):
-            self.reg_proj.export_file(record, field)
-
+        # self.reg_proj.delete_file(record, field)
```
Comment (sburns) [inline test/test_project.py]: Is this related to the tests not actually making any kind of side effect on the real server anymore?

File: test/test_project.py [hunk 2]
```diff
@@ -224,92 +495,110 @@ def test_export_survey_fields(self):
...
+    @responses.activate
     def test_import_dataframe(self):
         """Test importing a pandas.DataFrame"""
+        self.add_normalproject_response()
+        self.add_long_project_response()
         df = self.reg_proj.export_records(format='df')
         # grrr coerce implicilty converted floats to str(int())
-        for col in ['matrix1', 'matrix2', 'matrix3', 'sex']:
-            df[col] = map(lambda x: str(int(x)) if pd.notnull(x) else '', df[col])
+        # for col in ['matrix1', 'matrix2', 'matrix3', 'sex']:
```
Comment (sburns) [inline test/test_project.py]: It's better just to remove old code than comment it out, that's why we have version control :)

File: test/test_project.py [hunk 3]
```diff
@@ -224,92 +495,110 @@ def test_export_survey_fields(self):
...
     def test_date_formatting(self):
         """Test date_format parameter"""
-
+        self.add_normalproject_response()
+    
         def import_factory(date_string):
             return [{'study_id': '1',
                      'dob': date_string}]
-
+
```
Comment (sburns) [inline test/test_project.py]: There's white space in your new lines, can you remove that?

**Comment by erikh360** [comment]
@sburns awesome! I'll do some cleanup and commit soon!!

`responses` is an awesome package and it's usually possible to do it much cleaner, but everything in redcap goes over the same api endpoint so it's a bit clunky.

File: test/test_project.py (hunk 1 above)
Comment (erikh360) [reply test/test_project.py]: Yes, as everything is mocked now.

**Comment by erikh360** [comment]
@sburns Hi, reminder about this 🙂

**Comment by erikh360** [comment]
Hi @sburns Thanks for the merge! 

Can you please create a new version? 🙂

**Comment by erikh360** [comment]
@sburns Reminder to please create a release. 🙂

**Comment by sburns** [comment]
Actually in the mean time, @erikh360 do you know how to install from github?

You should be able to do `pip install -e git+https://github.com/sburns/PyCap.git#egg=PyCap`

**Comment by erikh360** [comment]
@sburns ah yes thanks, that was my backup plan.

**Pull request merged**